proteus/WaveTools.h

```diff
@@ -406,7 +406,7 @@ namespace proteus
 
 
 
- inline double* __cpp_uWindow(double* U, double x[nDim], double x0[nDim], double t, double* t0, double* kDir, double* kAbs, double* omega, double* phi, double* amplitude, double mwl, double depth, int N,int Nw, double* waveDir, double* vDir, double* tanhF, double gAbs , bool fast)
+ inline void __cpp_uWindow(double* U, double x[nDim], double x0[nDim], double t, double* t0, double* kDir, double* kAbs, double* omega, double* phi, double* amplitude, double mwl, double depth, int N,int Nw, double* waveDir, double* vDir, double* tanhF, double gAbs , bool fast)
```

Comment:
@adimako I found this pernicious little bug in the optimized WaveTools.h code. My understanding is that you designed the _u* functions to have no return value but this one declared it as double*.  I checked over the module and didn't find any other cases, but you might want to take a quick look. It generates a seg fault when optimization is on for the newer compilers. @ejtovar I suspect this may be what causes random waves test to fail on the conda stacks. Note, I also found an issue like this in the optimized quadrilateral mesh generation.  With these bug fix commits I also reset the hashdist test compile options to -O2 to verify that all the tests run with full optimization on the hashdist stack.